server/leafnode.go

```diff
@@ -1057,7 +1058,7 @@ func (c *client) processLeafnodeInfo(info *Info) {
 						c.Debugf("Error adding JetStream domain mapping: %v", err)
 					}
 				}
-			} else if hasJSDomain {
+			} else if hasJSDomain && opts.JetStreamDomain != info.Domain {
```

Comment:
Do we need to treat `info.Domain` possibly empty (old server?) differently here or not?

Reply:
No for this one we want both empty or exact match.